### PR TITLE
[FLINK-28481][k8s] Bump the fabric8 k8s-client to 5.12.3

### PR DESCRIPTION
**Pull request by bzhaoopenstack**
community hits FLINK-28481 mentioned issue, which will cause a Multiple
K8S-server lose the leadership in fabric8 5.5.0.  So plan to bump the
fabric8 version from 5.5.0 to 5.12.3

## What is the purpose of the change

The current fabric8 kubernetes-client(5.5.0) will swallow the KubernetesClientException and then the next renewing could not work properly until reach the deadline. This will be a serious problem because one time failure of renewing leader annotation will cause leadership lost. And refer the [issue](https://github.com/fabric8io/kubernetes-client/issues/4246), the said issue got progressed in the latest version 5.12.3

## Brief change log

Upgrade the fabric8 version to 5.12.3 and the associated dependencies to the right versions.

## Verifying this change

  - regression test needed for the associated parts in flink-k8s component.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes)
  - The S3 file system connector: ( no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-kubernetes/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -14,27 +14,27 @@ This project bundles the following dependencies under the Apache Software Licens
 - com.squareup.okhttp3:logging-interceptor:3.14.9
```
Comment (SteNicholas) [inline flink-kubernetes/src/main/resources/META-INF/NOTICE:14]: Please check whether the `okhttp3 ` dependencies has any change.

**Comment by SteNicholas** [comment]
@bzhaoopenstack, thanks for your contribution. There are a few inconsistencies for NOTICE file. You could refer to [NOTICE](https://github.com/SteNicholas/flink-kubernetes-operator/blob/af4ab7459c46b3df3f72dead2a12a8dadc3c889d/flink-kubernetes-operator/src/main/resources/META-INF/NOTICE) to update the NOTICE file of this pull request.

**Comment by wangyang0918** [comment]
Using the following command to get the bundled dependencies and verified.

```
cd flink-kubernetes
mvn clean install -DskipTests | grep Including | grep -v org.apache.flink | cut -d' ' -f3 | sort | sed s/:jar//g >/tmp/3
```

```
com.fasterxml.jackson.core:jackson-annotations:2.13.2
com.fasterxml.jackson.core:jackson-core:2.13.2
com.fasterxml.jackson.core:jackson-databind:2.13.2.2
com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.13.2
com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.13.2
com.squareup.okhttp3:logging-interceptor:3.14.9
com.squareup.okhttp3:okhttp:3.14.9
com.squareup.okio:okio:1.17.2
dk.brics.automaton:automaton:1.11-8
io.fabric8:kubernetes-client:5.12.3
io.fabric8:kubernetes-model-admissionregistration:5.12.3
io.fabric8:kubernetes-model-apiextensions:5.12.3
io.fabric8:kubernetes-model-apps:5.12.3
io.fabric8:kubernetes-model-autoscaling:5.12.3
io.fabric8:kubernetes-model-batch:5.12.3
io.fabric8:kubernetes-model-certificates:5.12.3
io.fabric8:kubernetes-model-common:5.12.3
io.fabric8:kubernetes-model-coordination:5.12.3
io.fabric8:kubernetes-model-core:5.12.3
io.fabric8:kubernetes-model-discovery:5.12.3
io.fabric8:kubernetes-model-events:5.12.3
io.fabric8:kubernetes-model-extensions:5.12.3
io.fabric8:kubernetes-model-flowcontrol:5.12.3
io.fabric8:kubernetes-model-metrics:5.12.3
io.fabric8:kubernetes-model-networking:5.12.3
io.fabric8:kubernetes-model-node:5.12.3
io.fabric8:kubernetes-model-policy:5.12.3
io.fabric8:kubernetes-model-rbac:5.12.3
io.fabric8:kubernetes-model-scheduling:5.12.3
io.fabric8:kubernetes-model-storageclass:5.12.3
io.fabric8:zjsonpatch:0.3.0
org.yaml:snakeyaml:1.27
```

**Review by wangyang0918 — APPROVED** [review]
+1 for merging.

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Pull request merged**